src/basic/string-util.h

```diff
@@ -228,3 +228,31 @@ static inline void *memory_startswith(const void *p, size_t sz, const char *toke
 
         return (uint8_t*) p + n;
 }
+
+/* Like startswith_no_case(), but operates on arbitrary memory blocks */
+static inline void *memory_startswith_no_case(const void *p, size_t sz, const char *token) {
+        size_t n;
+        char *p_lower, *token_lower;
+
+        assert(token);
+
+        n = strlen(token);
+        if (sz < n)
+                return NULL;
+
+        assert(p);
+
+        p_lower = ascii_strlower(strdup(p));
```

Comment:
You can't use strdup on arbitrary bytes - they might not contain any string terminator at all, so you'll need to use malloc + memcpy. And to use ascii_strlower I suppose you'll need to terminate the copy of the bytes with a zero byte yourself, so you will need to allocate sz+1 bytes to do that.

Reply:
Thanks for the review! You are right.

So I've come up with `strndup()`. According to the manpage:

```
If s is longer than n, only n  bytes are copied, and a terminating null byte ('\0') is added.
```

So we can assume that buffers are null-terminated before `ascii_strlower()`.

Also added additional NULL checks, as well as length variables `p_n`, `token_n`.

Reply:
@dongsupark thank you for the patch! I was wondering if it would be better to just iterate over `p` at most `sz` times calling `ascii_tolower` along the way.

Could you also add a few tests for `memory_startswith_no_case` to `src/test/test-string-util.c` right after `test_memory_startswith`? Something like `assert_se(memory_startswith_no_case((char[2]){'a', 'b'}, 2, "ab"))` should help catch the issue @krnowak pointed out (ASan would have reported a`stack-buffer-overflow` in this case).

Reply:
@evverx Done.

```
$ cd build ; meson test test-string-util
1/1 test-string-util                        OK       0.01 s 

OK:         1
FAIL:       0
SKIP:       0
TIMEOUT:    0
```